pandas/tseries/frequencies.py

```diff
@@ -144,6 +154,24 @@ def to_offset(freq) -> Optional[DateOffset]:
         except ValueError as err:
             raise ValueError(libfreqs.INVALID_FREQ_ERR_MSG.format(freq)) from err
 
+    elif isinstance(freq, int):
```

Comment:
As @jorisvandenbossche mentioned in the original https://github.com/pandas-dev/pandas/issues/9314#issuecomment-394330086, I'd be -0.5 accepting `int`s here.

e.g. `to_offset(0)` returning a `Year` offset is a bit opaque.